### PR TITLE
Add Tailscale Integration Docs

### PR DESCRIPTION
**Pull request by aj-foster**
This PR adds a new page to our Integrations section for CodeSandbox Repositories. As an _integration partner_ with Tailscale, we will provide documentation on how users can leverage Tailscale inside of a CodeSandbox Repository environment.

This is iteration one of the documentation. Future work may be done to create a deeper integration, which will require changes here.

The new page can be previewed [here](https://projects-docs-git-aj-tailscale-codesandbox1.vercel.app/docs/learn/integrations/tailscale).

### CODE REVIEW

File: packages/projects-docs/pages/learn/integrations/tailscale.mdx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import Callout from 'nextra-theme-docs/callout'
+
+# Tailscale
+
+At CodeSandbox, we use [Tailscale](https://tailscale.com/) to manage network access for private resources, including some of the services that make up CodeSandbox.
+As you work on an application in CodeSandbox Repositories, you might want to grant it access to private resources on your network.
+Tailscale can help.
+
+<Callout emoji="⭑">
+You can try Tailscale for free by following their [Quickstart guide](https://tailscale.com/kb/1017/install/#step-2-add-a-machine-to-your-network).
```
Comment (mayakacz) [inline packages/projects-docs/pages/learn/integrations/tailscale.mdx:15]:
```suggestion
You can try Tailscale for free by following their [Quickstart guide](https://tailscale.com/kb/1017/install/).
```

File: packages/projects-docs/pages/learn/integrations/tailscale.mdx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+Throughout this documentation, we'll assume you have a tailnet called `example-tailnet.ts.net` (which you should replace with your [name](https://tailscale.com/kb/1217/tailnet-name/)).
+</Callout>
+
+## Motivation
+
+Many applications exist as part of a collection of services.
+While things like mocks can help isolate a single application during development, this isn't always possible.
```
Comment (mayakacz) [inline packages/projects-docs/pages/learn/integrations/tailscale.mdx:22]: I don't understand this sentence.

File: packages/projects-docs/pages/learn/integrations/tailscale.mdx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+Just because you develop in CodeSandbox, however, doesn't mean you can't connect your application to private services running in your organization's network.
+
+Integrating with Tailscale provides containers running in a CodeSandbox Repository environment with access to your private resources.
+Combined with Tailscale's Magic DNS and ACLs, you can develop your application as if it were hosted in the same network as your other services — without compromising security.
```
Comment (mayakacz) [inline packages/projects-docs/pages/learn/integrations/tailscale.mdx:26]: Also consider linking to https://tailscale.com/kb/1081/magicdns/ and https://tailscale.com/kb/1018/acls/
```suggestion
Combined with Tailscale's MagicDNS and ACLs, you can develop your application as if it were hosted in the same network as your other services — without compromising security.
```

File: packages/projects-docs/pages/learn/integrations/tailscale.mdx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+## Creating an Auth Key
+
+Integrating a CodeSandbox Repository with Tailscale requires an [auth key](https://tailscale.com/kb/1085/auth-keys/), a secret you embed in the CodeSandbox environment to allow registration on your network.
+Network admins can create them on the [auth key page](https://login.tailscale.com/admin/settings/keys).
```
Comment (mayakacz) [inline packages/projects-docs/pages/learn/integrations/tailscale.mdx:31]: "Network Admin" is a specific role in Tailscale. "Admin" is more general.
```suggestion
Admins can create them on the [auth key page](https://login.tailscale.com/admin/settings/keys) of the Tailscale admin console.
```

File: packages/projects-docs/pages/learn/integrations/tailscale.mdx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+### Environment Variables
+
+Before launching the new container, it is necessary to add the Tailscale auth key as an environment variable.
+With a branch open in the Repository editor, go to the CodeSandbox menu, and choose **Env variables**.
+Throughout the rest of this documentation, we will assume you named the variable `TS_AUTH_KEY`.
+Restart any existing branches to update the running environment.
+
+### Docker Container
+
+In a setup using Docker Compose, we can add the `tailscale` image as an additional service beside our application.
```
Comment (mayakacz) [inline packages/projects-docs/pages/learn/integrations/tailscale.mdx:57]: Link to https://hub.docker.com/r/tailscale/tailscale

File: packages/projects-docs/pages/learn/integrations/tailscale.mdx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    command: "tailscaled --state=mem:"
+    volumes:
+      - "/dev/net/tun:/dev/net/tun"
+
+  myapp:
+    # ...
+    network_mode: service:tailscale
+```
+
+This runs the latest `tailscale/tailscale` image with a few customizations.
```
Comment (mayakacz) [inline packages/projects-docs/pages/learn/integrations/tailscale.mdx:77]: Here you call it `tailscale/tailscale` but earlier you just call it `tailscale`. Pick one.

**Comment by filipelima18** [comment]
This reads super well @aj-foster! Everything looks good to me.

**Pull request merged**